Translation/swedish

kumaheiyama opened the pull request:
Complete translation of pages into Swedish
Added folder content/sv

memark commented:
@raul 
Any update on this? Is this repo active?

Really looking forward to the Swedish translation!

mddarmawan commented:
Thank you @kumaheiyama! We'd love to have one reviewer per translation if we can - do you happen to have a colleague that can proof-read your PR? Colleague with same native language is preferred.

kumaheiyama commented:
> We'd love to have one reviewer per translation if we can - do you happen to have a colleague that can proof-read your PR? Colleague with same native language is preferred.

Hi,
No sorry, no one available. Maybe @memark or @dignite?

dignite commented:
Sure, I can proof read, swedish is my native language. I think I will get time to do this during the weekend.

mddarmawan commented:
Thanks, @dignite! Do you have any exact date estimation? Looking forward to it.

dignite commented on content/sv/who.md:
```diff
@@ -0,0 +1,4 @@
+Vem bör läsa detta dokument?
+==============================
+
+Alla utvecklare som utvecklar applikationer som körs som tjänst. DevOps och systemansvariga som produktionssätter eller hanterar sådana applikationer.
```
```suggestion
Alla utvecklare som utvecklar applikationer som körs som tjänst (Software as a Service). DevOps och systemansvariga som produktionssätter eller hanterar sådana applikationer.
```

"körs som tjänst" does not directly associate with Software as a Service for me (even though it is correct), could clarify with my suggestion?

dignite commented on content/sv/toc.md:
```diff
@@ -0,0 +1,38 @@
+## [VIII. Concurrency (Samtidighet)](./concurrency)
+### Skala ut systemet inom processmodellen
+
+## [IX. Disposability (Engångsbruk)](./disposability)
```
```suggestion
## [IX. Disposability (Till förfogande)](./disposability)
```

I wonder if ["Till förfogande"](https://sv.bab.la/lexikon/svensk-engelsk/till-f%C3%B6rfogande) is better here, as Engångsbruk translates to "One-time-use" and is not completely in line with how I view disposability. I could be wrong here though.

kumaheiyama replied:
I don't think "till förfogande" is quite the right translation in this context but how do you feel about "Tillgänglighet"? It would also align well with "Samtidighet" above.
Another possible applicable translation would be "Tillgängliggörande".

dignite replied:
Both those work, prefer "Tillgängliggörande" due to "Tillgänglihet" tying so close to availability which I connect to uptime.

kumaheiyama replied:
In each separate file we use the Swedish word first, and the English word in parentheses. Should we have that in the TOC too? What do you think?

kumaheiyama replied:
I'll make sure that all the subtitles are aligned with the subtitles in each file.

dignite commented on content/sv/toc.md:
```diff
@@ -0,0 +1,38 @@
+### Maximimera robusthet med snabb uppstart och smidig nedstängning
+
+## [X. Dev/prod parity (Likhet mellan miljöer)](./dev-prod-parity)
+### Låt utvecklings-, acceptanstests- och produktionsmiljö vara så lika som möjligt
```
```suggestion
### Låt utvecklingsmiljö, testmiljöer och produktionsmiljö vara så lika som möjligt
```

dignite commented on content/sv/toc.md:
```diff
@@ -0,0 +1,38 @@
+### Låt utvecklings-, acceptanstests- och produktionsmiljö vara så lika som möjligt
+
+## [XI. Logs (Loggar)](./logs)
+### Behandla logga som flöden av händelser
```
```suggestion
### Behandla loggar som flöden av händelser
```

dignite replied:
Or

```suggestion
### Behandla loggmeddelanden som flöden av händelser
```

dignite commented on content/sv/toc.md:
```diff
@@ -0,0 +1,38 @@
+### Behandla logga som flöden av händelser
+
+## [XII. Admin processes (Administrativa processer)](./admin-processes)
+### Köra administrativa uppgifter som engångsprocesser
```
```suggestion
### Kör administrativa uppgifter som engångsprocesser
```

dignite reviewed (commented):
> Thanks, @dignite! Do you have any exact date estimation? Looking forward to it.

I can't give a date estimation unfortunately but I can give updates on my progress. I have checked 7/19 files so far and I think it looks mostly great! Three nitpick comments so far and 2 typos.

kumaheiyama commented:
> Three nitpick comments so far and 2 typos.

Good work. Don't worry about being nitpicky, keep 'em coming! ;-)

dignite commented on content/sv/codebase.md:
```diff
@@ -0,0 +1,17 @@
+## I. Codebase (Kodbas)
```
The translations have either translated the heading completely or translated the heading and added ` (Codebase)` after.

```suggestion
## I. Kodbas
```

```suggestion
## I. Kodbas (Codebase)
```

This is the same for all 12 factors

dignite commented on content/sv/codebase.md:
```diff
@@ -0,0 +1,17 @@
+## I. Codebase (Kodbas)
+### En kodbas med versionshantering, flera produktionssättningar
+
+En tolvfaktorsapplikation hålls alltid i någon form av versionskontrollsystem, som exempelvis [Git](http://git-scm.com/), [Mercurial](https://www.mercurial-scm.org/), eller [Subversion](http://subversion.apache.org/). En kopia av versionshanteringsdatabasen kallas för ett *kodrepository*, ofta förkortat till *kodrepo* eller just *repo*.
```
```suggestion
En tolvfaktorsapplikation hålls alltid i någon form av versionshanteringssystem, som exempelvis [Git](http://git-scm.com/), [Mercurial](https://www.mercurial-scm.org/), eller [Subversion](http://subversion.apache.org/). En kopia av versionshanteringsdatabasen kallas för ett *kodrepository*, ofta förkortat till *kodrepo* eller just *repo*.
```

dignite commented on content/sv/codebase.md:
```diff
@@ -0,0 +1,17 @@
+
+En tolvfaktorsapplikation hålls alltid i någon form av versionskontrollsystem, som exempelvis [Git](http://git-scm.com/), [Mercurial](https://www.mercurial-scm.org/), eller [Subversion](http://subversion.apache.org/). En kopia av versionshanteringsdatabasen kallas för ett *kodrepository*, ofta förkortat till *kodrepo* eller just *repo*.
+
+En *kodbas* är varje enskilt repo (i ett centraliserat versionshanteringssystem såsom Subversion), eller varje set av repon som delar en viss koduppladdning som bas (i ett decentraliserat versionshanteringssystem såsom Git).
```
```suggestion
En *kodbas* är varje enskilt repo (i ett centraliserat versionshanteringssystem såsom Subversion), eller varje set av repon som delar en viss kodincheckning som bas (i ett decentraliserat versionshanteringssystem såsom Git).
```

dignite commented on content/sv/codebase.md:
```diff
@@ -0,0 +1,17 @@
+* Om det finns multipla kodbaser, är det inte en applikation -- det är ett distribuerat system. Varje komponent i ett distribuerat system är sin egen applikation, och kan således i sin egen rätt följa tolvfaktor-modellen.
+* Multipla applikationer som delar samma kod bryter mot tolvfaktor-modellen. Lösningen är att flytta ut delad kod till bibliotek som kan inkluderas genom [beroenden](./dependencies).
+
+Det finns alltid bara en kodbas per applikation, men det kommer bli många produktionssättningar av samma applikation. En *produktionssättning* är en körbar instans av applikationen. Det kan exempelvis vara en produktionssajt, eller en eller flera sajter för acceptanstestning. Dessutom har ofta respektive utvecklare en kopia av applikationen som kan köras lokalt i den egna utvecklingsmiljön, vilke ...
```
```suggestion
Det finns alltid bara en kodbas per applikation, men det kommer bli många produktionssättningar av samma applikation. En *produktionssättning* är en körbar instans av applikationen. Det kan exempelvis vara en produktionssajt, eller en eller flera sajter för testning. Dessutom har ofta respektive utvecklare en kopia av applikationen som kan köras lokalt i den egna utvecklingsmiljön, vilket också räknas som en form av produktionssättning.
```

I don't think we need to specifically mention acceptance testing. There is no good word matching "staging", in swedish speech the word staging is used but is not a swedish word. Produktionssättning roughly translates to production placement and is the word used when deploying to production. More general words used in swedish are "release" and "deploy", possibly "utrullning"(roll out). Therefore I suggest adding "en form av produktionssättning", "a shape/type of production deploy".

dignite commented on content/sv/codebase.md:
```diff
@@ -0,0 +1,17 @@
+
+Det finns alltid bara en kodbas per applikation, men det kommer bli många produktionssättningar av samma applikation. En *produktionssättning* är en körbar instans av applikationen. Det kan exempelvis vara en produktionssajt, eller en eller flera sajter för acceptanstestning. Dessutom har ofta respektive utvecklare en kopia av applikationen som kan köras lokalt i den egna utvecklingsmiljön, vilke ...
+
+Kodbasen är densamma för samtliga produktionssättningar, även om olika versioner kan vara gällande för varje enskild produktionsssättning. Till exempel, en utvecklare kan ha kodändringar som fortfarande inte har lagts över för acceptanstestning; acceptansmiljön kan ha kodändringar som ännu inte lagts över till produktion. Men alla delar samma kodbas, vilket gör dem identifierbara som olika versio ...
```
```suggestion
Kodbasen är densamma för samtliga produktionssättningar, även om olika versioner kan vara gällande för varje enskild produktionssättning. Till exempel kan en utvecklare ha kodändringar som ännu inte finns i någon testmiljö; testmiljöer kan ha kodändringar som ännu inte lagts över till produktion. Men alla delar samma kodbas, vilket gör dem identifierbara som olika versioner av samma applikation.
```

kumaheiyama replied:
I agree with the change suggestion, however I would like to change the semicolon to a colon as that makes more sense in context.

dignite reviewed (commented):
8/19 files checked.

Here comes Codebase.md review

dignite commented on content/sv/dependencies.md:
```diff
@@ -0,0 +1,12 @@
+
+De flesta programmeringsspråk erbjuder någon form av paketeringssystem för distribuering av stödbibliotek, som ex. [CPAN](http://www.cpan.org/) för Perl eller [Rubygems](http://rubygems.org/) för Ruby. Bibliotek installerade genom ett paketeringssystem kan installeras för hela arbetsmiljön (kallat för "systempaket", eng. "site packages") eller inom applikationens lokala utrymmet (kallat för "tred ...
+
+**En tolvfaktorapplikation förlitar sig aldrig på en implicit tillgänglighet av systempaket.** Den deklararerar alla beroenden, komplett och exakt, via ett *manifest för beroende-deklaration* (eng. "dependency declaration manifest"). Vidare använder den verktyg för *isolering av beroenden* under exekvering för att försäkra sig om att inga implicita beroenden "läckt in" från det omgivande körsyste ...
```
```suggestion
**En tolvfaktorapplikation förlitar sig aldrig på en implicit tillgänglighet av systempaket.** Den deklararerar alla beroenden, komplett och exakt, via ett *beroendespecifikationsmanifest* (eng. "dependency declaration manifest"). Vidare använder den verktyg för *isolering av beroenden* under exekvering för att försäkra sig om att inga implicita beroenden "läckt in" från det omgivande körsystemet. Den fullständiga och exakta beroendespecifikationen appliceras likadant på både produktionsmiljön och utvecklingsmiljön.
```

kumaheiyama replied:
Ouph, "beroendespecificationsmanifest" is a mouthful :-)

dignite commented on content/sv/dependencies.md:
```diff
@@ -0,0 +1,12 @@
+
+**En tolvfaktorapplikation förlitar sig aldrig på en implicit tillgänglighet av systempaket.** Den deklararerar alla beroenden, komplett och exakt, via ett *manifest för beroende-deklaration* (eng. "dependency declaration manifest"). Vidare använder den verktyg för *isolering av beroenden* under exekvering för att försäkra sig om att inga implicita beroenden "läckt in" från det omgivande körsyste ...
+
+Till exempel, [Bundler](https://bundler.io/) för Ruby erbjuder manifestformatet `Gemfile` för beroendedeklaration och `bundle exec` för beroendeisolering. I Python finns det två separata verktyg för dessa steg -- [Pip](http://www.pip-installer.org/en/latest/) används för deklaration och [Virtualenv](http://www.virtualenv.org/en/latest/) för isolering. Även C har [Autoconf](http://www.gnu.org/s/au ...
```
```suggestion
Som exempel så används [Bundler](https://bundler.io/) för Ruby erbjuder manifestformatet `Gemfile` för beroendespecifikation och `bundle exec` för beroendeisolering. I Python finns det två separata verktyg för dessa steg -- [Pip](http://www.pip-installer.org/en/latest/) används för beroendespecifikation och [Virtualenv](http://www.virtualenv.org/en/latest/) för isolering. Även C har [Autoconf](http://www.gnu.org/s/autoconf/) för beroendespecifikation, och statisk länkning kan användas som beroendeisolering. Oavsett vilka verktyg som används måste man använda beroendespecifikation och beroendeisolering tillsammans -- bara den ena eller den andra är inte tillräckligt för att tillfredsställa tolvfaktormodellen.
```

kumaheiyama replied:
I agree with the change to "beroendespecifikation", however language needs to be adjusted somewhat. Am I able to make a countersuggestion somehow? Changes marked with bold and italics.

_**Exempelvis**_ används [Bundler](https://bundler.io/) för Ruby **_som_** erbjuder manifestformatet `Gemfile` för beroendespecifikation och `bundle exec` för beroendeisolering. I Python finns det två separata verktyg för dessa steg -- [Pip](http://www.pip-installer.org/en/latest/) **_som_** används för beroendespecifikation och [Virtualenv](http://www.virtualenv.org/en/latest/) **_som används_** för isolering. Även C har [Autoconf](http://www.gnu.org/s/autoconf/) för beroendespecifikation, och statisk länkning kan användas som beroendeisolering. Oavsett vilka verktyg som används måste man använda beroendespecifikation och beroendeisolering tillsammans -- bara den ena eller den andra är inte tillräckligt för att tillfredsställa tolvfaktormodellen.

dignite replied:
Excellent improvements 👍

dignite commented on content/sv/dependencies.md:
```diff
@@ -0,0 +1,12 @@
+
+En fördel av att explicit deklararera beroenden är att det förenklar uppsättning av arbetsmiljön för nya utvecklare. Utvecklaren kan hämta hem kodbasen till sin egen dator, de enda förutsättningarna är att körrutiner (eng. "runtime") för utvecklingsspråket och hanteraren för beroenden finns förinstallerade. Dessa två kommer kunna sätta upp allting som behövs för att köra applikationens kod med et ...
+
+Tolvfaktorapplikationer förlitar sig heller inte på implicit tillgänglighet av vissa systemverktyg, ex. att skalverktyg som ImageMagick eller `curl` finns. Även om dessa verktyg kan finnas på många eller till och med de flesta system är det ingen garanti för att det finns på alla system där applikationen kan tänkas köras i framtiden, eller om versionen som finns installerad på systemet kommer var ...
```
```suggestion
Tolvfaktorapplikationer förlitar sig heller inte på implicit tillgänglighet av några systemverktyg, ex. att skalprogram som ImageMagick eller `curl` finns. Även om dessa verktyg kan finnas på många eller till och med de flesta system är det ingen garanti för att det finns på alla system där applikationen kan tänkas köras i framtiden, eller att versionen som finns installerad på systemet kommer vara kompatibel med applikationen. Om applikationen behöver anropa ett systemverktyg, bör det verktyget inkluderas som ett tredjepartspaket.
```

dignite reviewed (commented):
9/19 files checked.

Here comes dependencies.md review

dignite commented on content/sv/admin-processes.md:
```diff
@@ -0,0 +1,14 @@
+## XII. Admin processes (Administrativa processer)
+### Köra administrativa uppgifter som engångsprocesser
+
+[Process-skapandet](./concurrency) är en samling av processer som används för att utföra applikationens ordinarie verksamhet (som att exempelvis hantera webb-anrop) medan den är igång. Förutom detta vill utvecklare ofta utföra enskilda administrativa eller underhållsmässiga uppgifter för applikationen, som till exempel:
```
```suggestion
[Processformationen](./concurrency) är den samling processer som används för att utföra applikationens ordinarie verksamhet (som att exempelvis hantera webb-anrop) medan den är igång. Förutom detta vill utvecklare ofta utföra enskilda administrativa eller underhållsmässiga uppgifter för applikationen, som till exempel:
```

dignite commented on content/sv/concurrency.md:
```diff
@@ -0,0 +1,14 @@
+
+Detta utesluter inte att individuella processer hanterar deras egen interna arbetsfördelning, via trådar inuti den egna virtuella maskinen, eller via en asynkron-/händelse-modell som man hittar ex. i verktyg som [EventMachine](https://github.com/eventmachine/eventmachine), [Twisted](http://twistedmatrix.com/trac/), eller [Node.js](http://nodejs.org/). Men enskilda virtuella maskiner kan bara växa ...
+
+Processmodellen kommer verkligen till sin rätt när det gäller att skala upp. Modellens natur innebär att [man inte delar resurser och partionerar processer horisontellt](./processes) vilket betyder att lägga till mer samtidighet är en enkel och pålitlig operation. Utbudet av processtyper och antalet processer av respektive typ är vad som kallas för *processformering*.
```
```suggestion
Processmodellen kommer verkligen till sin rätt när det gäller att skala upp. Modellens natur innebär att [man inte delar resurser och partionerar processer horisontellt](./processes) vilket betyder att lägga till mer samtidighet är en enkel och pålitlig operation. Utbudet av processtyper och antalet processer av respektive typ är vad som kallas för *processformationen*.
```

dignite commented on content/sv/admin-processes.md:
```diff
@@ -0,0 +1,14 @@
+* Köra konsolapplikationer (även kända som [REPL skal](http://en.wikipedia.org/wiki/Read-eval-print_loop)) för att köra godtycklig kod eller undersöka en applikations modell mot en produktionsdatabas. De flesta språk erbjuder en REPL genom att köra kodtolken utan några argument (ex. `python` eller `perl`) eller har i vissa fall ett separat kommando (ex. `irb` i Ruby, `rails console` i Rails).
+* Köra enskilda engångsscript som lagts in i applikationens repo (ex. `php scripts/fix_bad_records.php`).
+
+Engångsprocesser bör köras i en identisk miljö som applikationens ordinarie [huvudprocess](./processes). De bör köras mot en [produktionsrelease](./build-release-run), som har samma [kodbas](./codebase) och samma [konfiguration](./config) som alla andra processer som är baserade på den releasen. Administrativ kod bör läggas tillsammans med applikationskoden för att undvika versionsproblem.
```
```suggestion
Engångsprocesser bör köras i en identisk miljö som applikationens ordinarie [processer](./processes). De bör köras mot en [produktionsrelease](./build-release-run), som har samma [kodbas](./codebase) och samma [konfiguration](./config) som alla andra processer som är baserade på den releasen. Administrativ kod bör läggas tillsammans med applikationskoden för att undvika versionsproblem.
```

dignite commented on content/sv/backing-services.md:
```diff
@@ -0,0 +1,14 @@
+
+Varje distinkt stödtjänst är en *resurs*. Till exempel, en MySQL-databas är en resurs; två MySQL-databaser (som används för lastbalansering av applikationslagret) kvalificeras som två distinkta resurser. Tolvfaktor-applikationen behandlar dessa databaser som *anslutna resurser*, vilket indikerar deras lösa koppling till den produkt som de är anslutna till.
+
+<img src="/images/attached-resources.png" class="full" alt="En produkt ansluten till fyra stödtjänster" />
```
```suggestion
<img src="/images/attached-resources.png" class="full" alt="En produktionssättning ansluten till fyra stödtjänster" />
```

dignite reviewed (commented):
Left to review:
- content/sv/build-release-run.md
- content/sv/concurrency.md
- content/sv/config.md
- content/sv/dev-prod-parity.md
- content/sv/logs.md
- content/sv/port-binding.md
- content/sv/processes.md

dignite commented on content/sv/config.md:
```diff
@@ -0,0 +1,22 @@
+* Inloggningsuppgifter mot externa tjänster såsom Amazon S3 eller Twitter
+* Produktionssättningsspecifika värden som exempelvis vedertagna värdnamn
+
+Applikationer sparar ibland konfiguration som konstanter i koden. Detta är ett brott mot tolvfaktor-modellen, som kräver en **strikt uppdelning mellan konfigration och kod**. Konfiguration varierar avsevärt mellan kodsläpp, kod gör det inte.
```
```suggestion
Applikationer sparar ibland konfiguration som konstanter i koden. Detta är ett brott mot tolvfaktor-modellen, som kräver en **strikt uppdelning mellan konfiguration och kod**. Konfiguration varierar avsevärt mellan kodsläpp, kod gör det inte.
```

dignite reviewed (commented):
14/19 files checked, will try to get to the last files tonight

kumaheiyama commented on content/sv/build-release-run.md:
```diff
@@ -0,0 +1,18 @@
+En [kodbas](./codebase) omvandlas till en produkt (avsedd för annat än utveckling) genom tre stadier:
+
+* *Byggstadiet* är en transformering som omvandlar ett kod-repo till ett exekverbart paket som kallas för ett *bygge* (eng. "build"). Byggprocessen använder sig av den version av koden som använts vid en viss uppladdning, hämtar tredjepartskomponenter som definierats som [beroenden](./dependencies) och kompilerar binära filer och resurser.
+* *Produktionssättningsstadiet* tar det som producerats vid byggstadiet och kombinerar det med den aktuella produktionssättningens [konfiguration](./config). Resultatet (kallad en *release*/*kodsläpp*) innehåller både byggpaketet och konfigurationen och är redo för omedelbar exekvering i den miljö som den är avsedd för.
```
I'll make we're using those terms everywhere, after you are done with the final file 👍🏻

dignite commented on content/sv/config.md:
```diff
@@ -0,0 +1,22 @@
+## III. Config (Konfiguration)
+### Lagra konfiguration i miljön
+
+En applikations *konfigration* berör allting som kan tänkas skilja sig mellan kodsläpp (acceptanstestnings-, produktions-, utvecklingsmiljöer etc). Detta inkluderar:
```
```suggestion
En applikations *konfiguration* berör allting som kan tänkas skilja sig mellan kodsläpp (acceptanstestnings-, produktions-, utvecklingsmiljöer etc). Detta inkluderar:
```

dignite reviewed (approved):
Good work! Bra jobbat :)

kumaheiyama commented:
@dignite Are you finished with all files?

dignite commented:
> @dignite Are you finished with all files?

Yes, sorry for the delayed response. I am finished, looks good.

kumaheiyama commented:
@dignite Done!
* I've updated the TOC to match all pages
* Replaced "produktionssättning" in most places with "kodsläpp" or "release". I did however keep it in a few places because the feeling was more formal and it seemed to fit.

Please have a final look at the latest commit and see if you have any comments, otherwise I think we are done.
Good work reviewing!

dignite commented on content/sv/backing-services.md:
```diff
@@ -0,0 +1,14 @@
+## IV. Stödstjänster (Backing services)
```
```suggestion
## IV. Stödtjänster (Backing services)
```

dignite commented on content/sv/build-release-run.md:
```diff
@@ -0,0 +1,18 @@
+## V. Bygga, släppa, köra (Build, deploy, run)
```
```suggestion
## V. Bygg, släpp, kör (Build, deploy, run)
```

dignite commented on content/sv/toc.md:
```diff
@@ -0,0 +1,38 @@
+## [III. Konfiguration (Config)](./config)
+### Lagra konfiguration i miljön
+
+## [IV. Stödstjänster (Backing services)](./backing-services)
```
```suggestion
## [IV. Stödtjänster (Backing services)](./backing-services)
```

dignite commented on content/sv/toc.md:
```diff
@@ -0,0 +1,38 @@
+## [IV. Stödstjänster (Backing services)](./backing-services)
+### Behandla stödstjänster som anslutna resurser
+
+## [V. Bygga, släppa, köra (Build, deploy, run)](./build-release-run)
```
```suggestion
## [V. Bygg, släpp, kör (Build, deploy, run)](./build-release-run)
```

dignite reviewed (approved):
Great work @kumaheiyama! I did notice a typo (Stödstjänster) and suggested a change in one of the headings to [imperativ](https://sv.wikipedia.org/wiki/Imperativ) (Bygg, släpp, kör) to match the english heading. This is the last feedback, I think the translation is great and will be a good addition.

kumaheiyama commented:
@dignite, I've corrected the last comments. Thanks for catching those!
@mddarmawan, I think that would conclude the review of the PR. Please let me know if there is anything else.

dignite commented:
Would love to see this merged after all the work put in by @kumaheiyama 

What is pending for this to be approved @mddarmawan ?

kumaheiyama commented:
@mddarmawan Ping

mddarmawan commented:
Sorry, I've missed this PR.

What great work, @dignite and @kumaheiyama! I am pushing it to Raul so it can be merged. Thanks for the contribution!

kumaheiyama commented:
@mddarmawan Still nothing?
